### PR TITLE
Added support for .rscignore file

### LINKED ISSUE #368: FR: create a .rscIgnore files to ignore files from being bundled
It is already possible to provide a `appFileManifest` in `deployApp` to list the files that should be bundle. 

I am often in the case like with `.gitignore` or `.Rbuildignore` : it is easier to tell what we don't want at all instead of listing everything we want. 

I would find it interesting to have a feature like that, a `.rscignore` file that would allow to always exclude some files to be bundled by `rsconnect`. 
This is like `.Rbuildignore` that prevent bundling in R CMD build.

We encountered this need in some scenarios where 
* we have projects that push in RStudio Connect 
* the development is done externally and the code is delivered in a gitlab. when tagged, it should be published in one of the url
* One person of the project in charge of deployment need to publish. This require to recreate the dev env and publish the file. Some files used in development do not need be be published but is is not easy to exclude them. 

Publishing unneeded file is not a problem, but can cause issues because they are scanned for dependencies and can increase the bundle size. 

What do you think ? 

In a way this issue relates to #342 about how to manage bundle files.

### PR DESCRIPTION
**Pull request by andyquinterom**
Resolves #368 

This pull request adds a simple .rscignore file that lists the files that the user wished to ignore when writing a manifest or deploying the app. 

I don't know where to write documentation, it is quite simple so if anyone can help me out, it would be appriciated.

Looking forward to your responses.

### CODE REVIEW

**Review by jmcphers — CHANGES_REQUESTED** [review]
Seems pretty straightforward, but does need documentation. I'd suggest adding this to the documentation for `deployApp` since it affects what is bundled during deployment. 

https://github.com/rstudio/rsconnect/blob/0534c51021fced67b81715e53ebc8abbc913fcad/R/deployApp.R#L9-L12

It should also be mentioned in the docs for `listBundleFiles`: 

https://github.com/rstudio/rsconnect/blob/0534c51021fced67b81715e53ebc8abbc913fcad/R/bundle.R#L178

**Comment by jmcphers** [comment]
Also - please not in documentation that this file does *not* support wildcards. No need to try to add support, but most ignore files do, so people may expect it unless indicated otherwise.

**Comment by andyquinterom** [comment]
I have just documented the use of the .rscignore file. If you see any opportunity for improvement let me know I'll make the appropriate changes.

File: R/deployApp.R [hunk 1]
```diff
@@ -4,6 +4,10 @@
 #' [RMarkdown][rmarkdown::rmarkdown-package] document, a plumber API, or HTML
 #' content to a server.
 #'
+#' In order to ignore specific files in the working directory while deploying,
+#' the files must be listed in the .rscignore file. This file must have one
+#' file or directory per line with no support for wildcards.'
+#'
```
Comment (jmcphers) [inline R/deployApp.R:10]: I think this should be moved to a less prominent location, but I'll take care of that after merge.

**Review by jmcphers — APPROVED** [review]
Very much appreciate the contribution!

**Comment by yogat3ch** [comment]
Hi folks, thanks for adding this functionality to rsconnect - it will be very helpful once I figure out how it works!
Here are my newb questions (I'd be happy to add the answers to these to the docs):
1. Where does the _.rscignore_ file go? In the top level project directory with the other ignore files? Or within one of the _rsconnect_ subdirectories?
2. When the _.rscignore_ file is properly place and working - how do we know? 
 2a. Will it remove the checkbox entries for the ignored files in the UI rsconnect deploy dialog?
 2b. Will it remove the files from the UI file checkbox dialog altogether?
3. Or Is this only used when deploying rsconnect::deployApp?
 3a. If so, how do we know it's working?
 
 
 Thanks in advance for your patience and advice,
 Stephen

**Comment by mikuo0628** [comment]
I echo @yogat3ch's comments. 

`.rscignore` is indeed very useful in my attempt to pipeline deployment of a Shiny Doc. But I could really use some clarity on its placement and behavior. 

My project folder (containing several Shiny Docs/apps) is structured as such:
```
+---/proj/script
|         +---01_deploy.R
|         +---shiny_doc1
|             +---shiny_doc1.Rmd (runtime:shiny_prerender)
|             +---.rscignore
|             +---css
|             +---data
|             +---images
|             +---www
|             +---rsconnect
|             +---data_prep
|                 +---aggregate_data_and_place_in_data_folder.R
|         +---shiny_doc2
|             +---shiny_doc2.Rmd
|             +---similar_to_above
|         +---shiny_doc3
|             +---etc
|         +---.rscignore
```
Yes, there are 2 `.rscignore` files. One is located at the `proj` root, the other is in Shiny Doc working directory `shiny_doc1`.
The root `.rscignore` is contains one line, listing the folder to exclude from bundling/deployment:
```
data_prep
```
The Shiny Doc working dir `.rscignore` is empty.
It's set up this way because, from my playing around,

1. Having `.rscignore` only in the Shiny Doc working dir, I run into issue with `file(con, 'r'): cannot open file; no such file`.  Deploying using GUI button/checkbox dialog runs into error `Could not find files to deploy: Error occurred while executing method`
2. Having `.rscignore` only in proj root dir won't have the issues above, but files/folders to ignore aren't recognized.
3. Having `.rscignore` in both locations, one with files/folders listed, the other empty, will work. `rsconnect::listBundleFiles` and GUI deploy window will show the correct list of files to bundle.

I've yet to see what happens when I have additional `.rscignore` in other Shiny Doc working dirs...

To answer some of @yogat3ch's questions based on what I've seen:

> 2. When the _.rscignore_ file is properly placed and working - how do we know?
>    2b. Will it remove the files from the UI file checkbox dialog altogether?

Yes. The files/folders are no longer seen in the checkbox dialog.

>    3a. If so, how do we know it's working?

Hopefully by `rsconnect::listBundleFiles`$contents.

Session info
```
R version 4.0.5 (2021-03-31)
Platform: i386-w64-mingw32/i386 (32-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_Canada.1252  LC_CTYPE=English_Canada.1252    LC_MONETARY=English_Canada.1252
[4] LC_NUMERIC=C                    LC_TIME=English_Canada.1252    

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
 [1] rsconnect_0.8.27 lubridate_1.8.0  magrittr_2.0.3   forcats_0.5.1    stringr_1.4.0    dplyr_1.0.8     
 [7] purrr_0.3.4      readr_2.1.2      tidyr_1.2.0      tibble_3.1.6     ggplot2_3.3.5    tidyverse_1.3.2 

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8          here_1.0.1          lattice_0.20-41     zoo_1.8-9           packrat_0.8.1      
 [6] assertthat_0.2.1    rprojroot_2.0.2     digest_0.6.29       utf8_1.2.2          mime_0.12          
[11] R6_2.5.1            cellranger_1.1.0    backports_1.4.1     reprex_2.0.1        evaluate_0.15      
[16] httr_1.4.2          pillar_1.7.0        rlang_1.0.1         readxl_1.4.1        googlesheets4_1.0.0
[21] curl_4.3.2          rstudioapi_0.13     jquerylib_0.1.4     rmarkdown_2.13      googledrive_2.0.0  
[26] munsell_0.5.0       shiny_1.7.1         broom_0.7.12        compiler_4.0.5      httpuv_1.6.5       
[31] modelr_0.1.8        xfun_0.30           pkgconfig_2.0.3     askpass_1.1         htmltools_0.5.2    
[36] openssl_2.0.3       tidyselect_1.1.2    fansi_1.0.2         withr_2.4.3         crayon_1.5.0       
[41] tzdb_0.2.0          dbplyr_2.1.1        later_1.3.0         grid_4.0.5          jsonlite_1.8.0     
[46] xtable_1.8-4        gtable_0.3.0        lifecycle_1.0.1     DBI_1.1.2           scales_1.1.1       
[51] stringi_1.7.6       cli_3.2.0           renv_0.15.4         fs_1.5.2            promises_1.2.0.1   
[56] xml2_1.3.3          bslib_0.3.1         ellipsis_0.3.2      generics_0.1.2      vctrs_0.3.8        
[61] tools_4.0.5         glue_1.6.1          hms_1.1.1           fastmap_1.1.0       yaml_2.3.5         
[66] colorspace_2.0-2    gargle_1.2.0        rvest_1.0.2         gt_0.4.0            knitr_1.37         
[71] haven_2.4.3         sass_0.4.0         
```

**Comment by fernandoroa** [comment]
considering I deploy apps by command, I am having problem with making .rscignore work properly too. I tested several things without success. I came to a workaround not using .rscignore. Consider list.files normally ignores hidden files too (.*), this worked:

```
file_list <- list.files(recursive=TRUE)
file_list_rem_pattern <- grep("patternToRemove", file_list, invert=T, value=T)
rsconnect::deployApp(appName = "appname", account = "accountName", server = "serverUrl", appFiles = file_list_rem_pattern)
```

**Comment by yogat3ch** [comment]
That is a super helpful rundown of the functionality. Thank you for answering my questions! @mikuo0628

**Pull request merged**